components/raftstore/src/store/config.rs

```diff
@@ -754,34 +756,64 @@ impl Config {
     }
 }
 
-pub struct RaftstoreConfigManager(pub Arc<VersionTrack<Config>>);
+pub struct RaftstoreConfigManager {
+    scheduler: Scheduler<RefreshConfigTask>,
+    config: Arc<VersionTrack<Config>>,
+}
+
+impl RaftstoreConfigManager {
+    pub fn new(
+        scheduler: Scheduler<RefreshConfigTask>,
+        config: Arc<VersionTrack<Config>>,
+    ) -> RaftstoreConfigManager {
+        RaftstoreConfigManager { scheduler, config }
+    }
+}
 
 impl ConfigManager for RaftstoreConfigManager {
     fn dispatch(
         &mut self,
-        change: ConfigChange,
+        mut change: ConfigChange,
     ) -> std::result::Result<(), Box<dyn std::error::Error>> {
+        let change_str = format!("{:?}", change);
         {
             let change = change.clone();
-            self.0.update(move |cfg: &mut Config| cfg.update(change));
+            self.config
+                .update(move |cfg: &mut Config| cfg.update(change));
+        }
+        Config::write_change_into_metrics(change.clone());
```

Comment:
Why move this line forward?

Reply:
Because the following will remove elements from `change`. Will the preposition affect the accuracy?

Reply:
> Will the preposition affect the accuracy?

It won't

> Because the following will remove elements from change

How about `get().clone()`, so we don't need to clone the whole changes for metrics.

Reply:
Sure, fixed, thanks!